### PR TITLE
cringify: init at 1.1

### LINKED ISSUE #249956: Package request: cringify
**Project description**

Cringify is a cli tool to annoy your friends on the internet by transforming `Rust is for hippies` into `rUsT Is fOr hIpPiEs`

**Metadata**

* homepage URL: https://github.com/sansyrox/cringify
* source URL: https://github.com/sansyrox/cringify
* license: mit
* platforms: unix, linux, darwin.... (ANYTHING RUSTC SUPPORTS)

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/249956

Adds 1 new package: `cringify`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/cringify/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "cringify";
+  version = "1.1";
```
Comment (antonmosich) [inline pkgs/tools/text/cringify/default.nix:8]: It's a bit confusing with the version of this package. In the `cargo.toml` and `cargo.lock` files it's version is listed as 0.1.1 (not 1.1), while when calling `cringify -V` it tells me it's version is 0.1.0.
Either way, 1.1 seems to not be the correct version.

Comment (TomaSajt) [reply pkgs/tools/text/cringify/default.nix]: -V had to be patched, as upstream has not updated it

**Review by antonmosich — APPROVED** [review]
Looks good to me now :)

**Comment by matthiasbeyer** [comment]
Result of `nixpkgs-review pr 250196` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cringify</li>
  </ul>
</details>

**Pull request merged**